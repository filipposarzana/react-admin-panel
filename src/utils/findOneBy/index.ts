import { User, users } from '~/db'

type Key = Extract<keyof User, 'name' | 'uuid'>

const findOneBy = <K extends Key>(prop: K) => (value: User[K]) =>
  users.find((user) => user[prop].toLowerCase() === value.toLowerCase())

export const findOneByName = findOneBy('name')

export const findOneByUuid = findOneBy('uuid')
