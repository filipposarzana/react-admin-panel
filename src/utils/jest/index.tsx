import { render } from '@testing-library/react'
import { renderHook } from '@testing-library/react-hooks'
import { ReactElement } from 'react'
import { createRenderer } from 'react-test-renderer/shallow'
import { StyleSheetManager } from 'styled-components'

export { act, fireEvent, waitFor } from '@testing-library/react'

export const renderer = {
  deep: (node: ReactElement) => {
    const r = render(<StyleSheetManager disableVendorPrefixes>{node}</StyleSheetManager>)

    return {
      ...r,
      output: r.container.firstChild,
    }
  },
  hook: <T extends unknown, R extends unknown>(hookFunction: (props: T) => R) => {
    const r = renderHook(hookFunction)

    return r
  },
  shallow: (node: ReactElement) => {
    const r = createRenderer()

    r.render(node)

    return {
      instance: r.getMountedInstance(),
      output: r.getRenderOutput(),
      unmount: r.unmount,
    }
  },
}
