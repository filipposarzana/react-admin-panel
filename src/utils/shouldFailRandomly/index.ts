export const shouldFailRandomly = () => Math.round(Math.random()) === 0
