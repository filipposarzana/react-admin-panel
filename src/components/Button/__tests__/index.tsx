import { renderer } from '~/utils/jest'
import { Button, Props } from '..'

const fixtures: [string, Omit<Props, 'children'>][] = [
  ['matches snapshot with all props', { disabled: true, onClick: jest.fn(), type: 'submit' }],
  ['matches snapshot with default props', { type: 'button' }],
]

describe('Button', () => {
  fixtures.forEach(([title, props]) => {
    it(title, () => {
      const { output } = renderer.shallow(<Button {...props}>Save</Button>)

      expect(output).toMatchSnapshot()
    })
  })
})
