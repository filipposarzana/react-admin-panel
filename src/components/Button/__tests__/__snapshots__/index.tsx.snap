// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot with all props 1`] = `
<styled.div
  as="button"
  background="gray900"
  borderRadius={4}
  data-test-id="button"
  disabled={true}
  onClick={[MockFunction]}
  p={8}
  type="submit"
>
  <styled.span
    colorName="white"
    kind="bodyM"
  >
    Save
  </styled.span>
</styled.div>
`;

exports[`Button matches snapshot with default props 1`] = `
<styled.div
  as="button"
  background="gray900"
  borderRadius={4}
  data-test-id="button"
  disabled={false}
  onClick={[Function]}
  p={8}
  type="button"
>
  <styled.span
    colorName="white"
    kind="bodyM"
  >
    Save
  </styled.span>
</styled.div>
`;
